src/cli/commands/files/add.js

```diff
@@ -94,26 +107,62 @@ module.exports = {
       }
     }
 
-    createAddStream((err, addStream) => {
-      if (err) {
+    // if there are piped arguments, input is data to publish instead of a file
+    // path
+    if(hasPipedArgs) {
+      const data = argv.file
+      const dataStream = new stream.Readable()
+      dataStream.push(Buffer.from(data, 'utf8'))
+      dataStream.push(null)
+      createAddStream((err, addStream) => {
+        if(err)
         throw err
-      }
+        
+        addDataPipeline([dataStream], addStream)
+      })
```

Comment:
I see. So the right solution would be to do something like

`
if (hasStdin) {
  ipfs.files.add({path:'', content: process.stdin}, (err, files)
} else { 
// as it is now
}
`